libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Actions/Microsoft.EmitEvent.schema

```diff
@@ -14,53 +14,48 @@
             "description": "Optional id for the dialog"
         },
         "disabled": {
-            "$role": "expression",
-            "type": "boolean",
+            "$ref": "https://raw.githubusercontent.com/microsoft/botframework-sdk/master/schemas/component/definitions.schema#/definitions/booleanExpression",
             "title": "Disabled",
             "description": "Optional condition which if true will disable this action.",
             "examples": [
                 "user.age > 3"
             ]
         },
         "eventName": {
-            "$role": "expression",
-            "type": "string",
+            "$ref": "https://raw.githubusercontent.com/microsoft/botframework-sdk/master/schemas/component/definitions.schema#/definitions/stringExpression",
             "title": "Event name",
             "description": "Name of the event to emit.",
-            "enum": [
-                "beginDialog",
-                "resumeDialog",
-                "repromptDialog",
-                "cancelDialog",
-                "endDialog",
-                "activityReceived",
-                "recognizedIntent",
-                "unknownIntent",
-                "actionsStarted",
-                "actionsSaved",
-                "actionsEnded",
-                "actionsResumed"
+            "oneOf": [
+                {
+                    "enum": [
+                        "beginDialog",
+                        "resumeDialog",
+                        "repromptDialog",
+                        "cancelDialog",
+                        "endDialog",
+                        "activityReceived",
+                        "recognizedIntent",
+                        "unknownIntent",
+                        "actionsStarted",
+                        "actionsSaved",
+                        "actionsEnded",
+                        "actionsResumed"
+                    ]
+                },
+                {
+                    "pattern": "^(?!(beginDialog$|resumeDialog$|repromptDialog$|cancelDialog$|endDialog$|activityReceived$|recognizedIntent$|unknownIntent$|actionsStarted$|actionsSaved$|actionsEnded$|actionsResumed))(\\S){1}.*"
```

Comment:
>(\\S){1}.* [](start = 216, length = 10)

Regex

Reply:
Again I have tested this extensively.  What this combination does is to say the value needs to be one of the enum OR a string which is not that enum.

This is 100% to drive VSCode.  It gives us the options enumerated via the enum in vscode, but allows other values to be passed as well.

---
In reply to: [410953274](https://github.com/microsoft/botbuilder-dotnet/pull/3758#discussion_r410953274) [](ancestors = 410953274)